docs/fsharp-cheatsheet.md

```diff
@@ -555,136 +555,158 @@ Another way of implementing interfaces is to use *object expressions*.
 ### Modules
 Modules are key building blocks for grouping related code; they can contain `types`, `let` bindings, or (nested) sub `module`s.
-Identifiers within modules can be referenced using dot notation, or you can bring them into scope via the `open` keyword. Illustrative-only example:
-
-    module Game =
-        let mutable basePoints = 1
-        type Player = { id: int; score: int }
-        let playerScored player = { player with score = player.score + basePoints }
-
-    let player: Game.Player = { id = 1; score = 0 }
-    let player' = Game.playerScored player  // score = 1
-
-    open Game
-    basePoints <- 2
-    let player'' = playerScored player'  // player''.score = 3
-
-If you have only one module in a file, the `module` name can be declared at the top of the file, with all further declarations
-being module elements (and non indentation required)
+Identifiers within modules can be referenced using dot notation, or you can bring them into scope via the [`open`](#CodeOrganization_OpenAndAutoOpen) keyword.
+Illustrative-only example:
+
+    module Money =
+        type CardInfo =
+            { number: string
+              expiration: int * int }
+
+        type Payment =
+            | Card of CardInfo
+            | Cash of int
+
+        module Functions =
+            let getCreditCardType (cardNumber: string) =
+                match cardNumber.ToCharArray() |> Array.truncate 2 with
+                | [| '3'; '4' |] | [| '3'; '7' |]      -> "Amex"
+                | [| '4'; _ |]                         -> "Visa"
+                | [| '5'; x |] when '0' < x && x < '6' -> "MasterCard"
+                | [| '6'; x |] when x = '0' || x = '5' -> "Discover"
+                | _                                    -> "Unknown"
+
+If there is only one module in a file, the `module` name can be declared at the top, and all code constructs
+within the file will be included in the `module`s definition (no indentation required).
 
     module Functions  // notice there is no '=' when at the top of a file
 
-    let addToFive num = 5 + num
-    let subtractFive num = num - 5
+    let sumOfSquares n = seq {1..n} |> Seq.sumBy (fun x -> x * x)  // Functions.sumOfSquares
 
 ### Namespaces
-Namespaces are simply dotted names that prefix other program elements to allow for further hierarchical organization.
-All `type` and `module` elements that follow a `namespace` declaration will require an [`open`](#CodeOrganization_OpenAndAutoOpen) or to be dotted-into to access.
-If a new `namespace` is specified, all elements following will be part of the new namespace.
+Namespaces are simply dotted names that prefix `type` and `module` declarations to allow for hierarchical scoping.
+The first `namespace` directives must be placed at the top of the file. Subsequent `namespace` directives either:
+(a) create a sub-namespace (keeping prior declarations in scope); or (b) create a new namespace (and all prior declarations are discarded from scope).
```

Comment:
you sure there are subnamespaces (keeping decls in scope)?

Reply:
Dang it. You're right.